CMakeLists.txt

```diff
@@ -38,15 +38,51 @@ if ( ${SANITIZE} )
     add_link_options( -fsanitize=undefined,address -fno-omit-frame-pointer )
 endif ()
 
-# Quality of life external libs
-add_subdirectory ( 3rd-party/range-v3 EXCLUDE_FROM_ALL )
-add_subdirectory ( 3rd-party/spdlog EXCLUDE_FROM_ALL )
+function ( add_spirv_shader TARGET_NAME INPUT_FILE )
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv
+        COMMAND ${glslc} -c ${INPUT_FILE} -o ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv
+        MAIN_DEPENDENCY ${INPUT_FILE}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    )
+    add_custom_target ( ${TARGET_NAME} ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv )
+    set ( ${TARGET_NAME}_OUTPUTS ${CMAKE_CURRENT_BINARY_DIR}/${TARGET_NAME}.spv PARENT_SCOPE )
+endfunction ()
+
+function ( suppress_warnings TARGET_NAME )
+    get_target_property( target_options ${TARGET_NAME} COMPILE_OPTIONS )
+
+    if ( MSVC )
+        list( REMOVE_ITEM target_options /W4 )
+    else ()
+        list( REMOVE_ITEM target_options -Wall -Wextra -Wpedantic )
```

Comment:
To disable warnings from gcc and clang you can pass ```-w``` option.
https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
This way not only ```-Wall, e.t.c``` will get disabled

Reply:
It seems to be more clear. Agree.